src/build/__main__.py

```diff
@@ -55,7 +55,7 @@ def _build_in_current_env(builder, outdir, distributions, skip_dependencies=Fals
         if not skip_dependencies:
             missing = builder.check_dependencies(dist)
             if missing:
-                _error('Missing dependencies:' + ''.join(['\n\t' + dep for dep in missing]))
+                _error('Unmet dependencies:' + ''.join(['\n\t' + ' -> '.join(unmet_deps) for unmet_deps in missing]))
```

Comment:
Why change the message string here ("missing" -> "unmet")?

I am not really a fan of the way this is displayed in the CLI:
```
ERROR Unmet dependencies:
	apispec[validation] -> prance[osv] (>=0.11) ; extra == 'validation' -> openapi-spec-validator (>0.2,>=0.2.1) ; extra == 'osv'
```

I think the following would be better:
```
ERROR Unmet dependencies:
	apispec[validation]
        prance[osv] (>=0.11) -> openapi-spec-validator (>0.2,>=0.2.1)
```

Reply:
> Why change the message string here ("missing" -> "unmet")?

'Missing' might lead one to think that the dependency can't be located when its version is incompatible.

> I think the following would be better:

It's not clear that the unmet dependency here is `openapi-spec-validator`.  Maybe something like this would be better?

```
ERROR Unmet dependencies:
	openapi-spec-validator (>0.2,>=0.2.1) ; extra == 'osv'
		from prance[osv] (>=0.11) ; extra == 'validation'
				from apispec[validation]
```

Reply:
> 'Missing' might lead one to think that the dependency can't be located when its version is incompatible.

The requirement string would include the version requirement, so I find that very unlikely. I do not think this is reason enough to change.

> It's not clear that the unmet dependency here is `openapi-spec-validator`. Maybe something like this would be better?

Well, it is certainly not optimal but I think it's clear enough. Your proposed way does not work great from long chains, I see two issues with it: 1) it puts all dependencies on a new line, which works very badly on >5 chains, and 2) it puts the actual requirement in the end, which is hard to find. It also has the `; extra = ...` which is completely avoidable.

As a user, what I am most interested in knowing is the 1st level missing dependency, in this case `apispec[validation]`, so that I can `pip install apispec[validation]`. As a distro packager, I want to know the actual dependency in the tree, and that is clear enough to me in the format I proposed. We should optimize for the general user.

So, for the interface here I have the following requirements:
  - make it clear, and easily accessible, what is 1st level missing dependency (in this case `apispec[validation]`)
  - provide information about the missing dependency chain (but do not optimize for this)
  - make sure the output does not become unreadable on long chains

Reply:
I've changed the formatting as suggested.